LibreNMS/Alert/AlertRules.php

```diff
@@ -106,31 +109,50 @@ public function runRules($device_id)
                 $doalert = false;
             }
 
-            $current_state = dbFetchCell('SELECT state FROM alerts WHERE rule_id = ? AND device_id = ? ORDER BY id DESC LIMIT 1', [$rule['id'], $device_id]);
+            $current_state = Alert::where('rule_id', $rule['id'])
+                ->where('device_id', $device_id)
+                ->orderBy('id', 'desc')
+                ->limit(1)
+                ->value('state');
             if ($doalert) {
                 if ($current_state == AlertState::ACKNOWLEDGED) {
                     Log::info('Status: %ySKIP%n', ['color' => true]);
                 } elseif ($current_state >= AlertState::ACTIVE) {
                     Log::info('Status: %bNOCHG%n', ['color' => true]);
                     // NOCHG here doesn't mean no change full stop. It means no change to the alert state
                     // So we update the details column with any fresh changes to the alert output we might have.
-                    $alert_log = dbFetchRow('SELECT alert_log.id, alert_log.details FROM alert_log,alert_rules WHERE alert_log.rule_id = alert_rules.id && alert_log.device_id = ? && alert_log.rule_id = ? && alert_rules.disabled = 0
-     ORDER BY alert_log.id DESC LIMIT 1', [$device_id, $rule['id']]);
+                    $alert_log = AlertLog::leftJoin('alert_rules', 'alert_log.rule_id', '=', 'alert_rules.id')
+                        ->where('alert_log.device_id', $device_id)
+                        ->where('alert_log.rule_id', $rule['id'])
+                        ->where('alert_rules.disabled', 0)
+                        ->orderBy('alert_log.id', 'desc')
+                        ->first(['alert_log.id', 'alert_log.details']);
                     $details = [];
-                    if (! empty($alert_log['details'])) {
-                        $details = json_decode(gzuncompress($alert_log['details']), true);
-                    }
                     $details['contacts'] = AlertUtil::getContacts($qry);
                     $details['rule'] = $qry;
-                    $details = gzcompress(json_encode($details), 9);
-                    dbUpdate(['details' => $details], 'alert_log', 'id = ?', [$alert_log['id']]);
+                    AlertLog::where('id', $alert_log->id)
+                        ->update(['details' => $details]);
```

Comment:
```php
$details = $alert_log->details;
$details['contacts'] = AlertUtil::getContacts($qry);
$details['rule'] = $qry;
$alert_log->save();
```

Reply:
Note, you can't modify it directly here you can only assign a complete modified array.

Reply:
I'm not sure if fetching the model partially affects the ->save()